### PR TITLE
Add `NoRewindIterator` stub.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Hi, please just add one stub, make the build green, and wait for feedback.

**Comment by drupol** [comment]
Ok will split this in multiple PRs.

File: stubs/iterable.stub [hunk 1]
```diff
@@ -298,3 +296,28 @@ class AppendIterator extends IteratorIterator {
...
 }
+
+/**
+ * @template-covariant TKey
+ * @template-covariant TValue
+ * @template TIterator as Iterator<TKey, TValue>
+ *
+ * @template-extends IteratorIterator<TKey, TValue, TIterator>
+ */
+class NoRewindIterator extends IteratorIterator {
```
Comment (ondrejmirtes) [inline stubs/iterable.stub:307]: This stub is different from the one in Psalm: https://github.com/vimeo/psalm/blob/36d5a2a83c1d7b3e2aca1cfe9e9dcc916619e962/stubs/CoreGenericIterators.phpstub#L581-L604 (This one uses three template types, Psalm uses two). We need to be cross-compatible here.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Also - this class needs to be added to `parameters.featureToggles.skipCheckGenericClasses` in conf/config.neon.

**Comment by ondrejmirtes** [comment]
All the other PRs have exactly the same problems so I'm not gonna repeat the reviews there.

**Comment by ondrejmirtes** [comment]
(That's why I asked you to submit only one stub and wait for feedback.)

**Comment by drupol** [comment]
In the end, should `IteratorIterator` have 2 or 3 parameters? Thanks!

**Comment by ondrejmirtes** [comment]
In both Psalm and PHPStan it currently has three (https://github.com/vimeo/psalm/blob/36d5a2a83c1d7b3e2aca1cfe9e9dcc916619e962/stubs/CoreGenericIterators.phpstub#L104-L113) so I don't know why you think it needs to change?

**Comment by drupol** [comment]
Just to be sure, nothing else.

File: stubs/iterable.stub (hunk 1 above)
Comment (orklah) [reply stubs/iterable.stub:307]: Note that NoRewindIterator has three templates in the next major of Psalm: https://github.com/vimeo/psalm/blob/master/stubs/CoreGenericIterators.phpstub#L593

(as well as every child of IteratorIterator IIRC)

EDIT: more context here: https://github.com/vimeo/psalm/pull/6970 it originated from a change in phpstan that we duplicated in Psalm but only on our master branch

Comment (ondrejmirtes) [reply stubs/iterable.stub:307]: Oh, I didn't realize that. So I'll recheck the stubs against master.

Comment (drupol) [reply stubs/iterable.stub:307]: Indeed, I inspired those PRs from what's in PSalm (`@master` branch)

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**